server/src/controllers/newsletter.js

```diff
@@ -0,0 +1,46 @@
+//const User=require('../model/user')
+import User from '../model/user.js'
+//const nodemailer = require('nodemailer');
+import nodemailer from 'nodemailer';
+import dotenv from 'dotenv';
+dotenv.config();
+
+export const mailer= async (req,res)=>{
+    const users=await User.find({newsletterIsSubscribed:true}); // this is an array of objects
+    const mails=users.map(user=>user.email); // this is an array of emails
+
+    const transporter = nodemailer.createTransport({
+        service: 'gmail',
+        host: 'smtp.gmail.com',
+        port: 587,
+      secure: false, // Use `true` for port 465, `false` for all other ports
+      auth: {
+        user: String(process.env.USER),
+        pass: String(process.env.PASS),
```

Comment:
instead of user, pass, use email address, passkey

Reason: more descriptive on what these variable are doing, value passed